apis/hive/v1/clusterpool_types.go

```diff
@@ -187,6 +208,25 @@ const (
 	// ClusterPoolAllClustersCurrentCondition indicates whether all unassigned (installing or ready)
 	// ClusterDeployments in the pool match the current configuration of the ClusterPool.
 	ClusterPoolAllClustersCurrentCondition ClusterPoolConditionType = "AllClustersCurrent"
+	// ClusterPoolInventoryValidCondition is set to provide information on whether the cluster pool inventory is valid
+	ClusterPoolInventoryValidCondition ClusterPoolConditionType = "InventoryValid"
+)
+
+// Inventory (in)valid reasons
+const (
+	// InventoryReasonValid is used when all ClusterDeploymentCustomization are
+	// available and when used the ClusterDeployments are successfully installed
+	InventoryReasonValid = "Valid"
+	// InventoryReasonMissing is used when one or more ClusterDeploymentCustomization are missing
+	InventoryReasonMissing = "Missing"
+	// InventoryReasonFound is used cancel a missing ClusterDeploymentCustomization
+	InventoryReasonFound = "Found"
+	// InventoryReasonBrokenByCloud is used when one or more ClusterDeployments installations failed
+	InventoryReasonBrokenByCloud = "BrokenByCloud"
+	// InvenotryReasonBrokenBySyntax is used when one or more ClusterDeploymentCustomization patching failed
+	InvenotryReasonBrokenBySyntax = "BrokenBySyntax"
+	// InventoryReasonInvalid is used when multiple reasons and ClusterDeploymentCustomizations are incompatible
+	InventoryReasonInvalid = "Invalid"
```

Comment:
Looking at this again, I don't think we need this level of granularity on the **ClusterPool**, especially since there could be different issues with different CDCs. Those issues are all denoted in the `message`; so I think the Reason can just be "InventoryValid" or "InventoryInvalid". This should simplify the logic in `updateInventoryValidMessage`: basically if there's any problem you set `status=False, reason=InventoryInvalid`; otherwise `status=True, reason=InventoryValid`.

(I started off being concerned about the potential non-idempotence of "Found". That concern is also eliminated if you do the above.)